grpc/README.md

```diff
@@ -0,0 +1,333 @@
+```
+
+Open the file in the editor and add the following code to the Dockerfile:
+
+```sh
+vi Dockerfile
+```
+
+```
+FROM python:3.7
+
+WORKDIR /app
+
+COPY . .
+
+RUN pip install -r requirements.txt
+
+EXPOSE 8080
+
+CMD ["python", "greeter_server.py"]
+
+```
+
+
+Edit the requirements file in the editor and add the following lines:
+
+```sh
+vi requirements.txt
+```
+
+```
+grpcio
+protobuf
+```
+
+
+Build the Docker image and then list the images created:
+
+```sh
+docker build .
+docker images -a
+```
+
+
+Save the imageid from the previous step to an environment variable and run the Docker image locally:
+
+```sh
+export DOCKER_IMAGE="<YOUR_IMAGE_ID_FROM_ABOVE>"
+docker run -p 8080:8080 -it $DOCKER_IMAGE
+```
+
+
+Open up a 2nd cloud shell session or terminal tab. Use grpcurl to ensure the service is working. grpcurl expects to send Service.Method to gRPC endpoints and validates that against the proto file. If you look at the protofile referenced, you’ll see that the package keyword is used to specify the package name for the service. The package name is used to uniquely identify the service.  In this case, the fully qualified name of the service is helloworld.Greeter
+
+```sh
+grpcurl -plaintext -import-path $HOME/grpc/examples/protos -proto helloworld.proto -d '{"name":"Guest"}' localhost:8080 helloworld.Greeter/SayHello
+```
+
+If you get a response like the one below, the Docker image is working!:
+
+```
+{
+    "message": "Hello, Guest!"
+}
+```
+
+Once the image is working, you can close the 2nd tab. Let's go back to the first shell and stop the image using `Ctrl+C`. 
+Now let's create a global Artifact Registry and configure Docker to use the Artifact Registry:
+
+```sh
+gcloud artifacts repositories create grpcdemo \
+    --repository-format=Docker \
+    --location=us \
+    --description="gRPC Demo repo"
+
+gcloud auth configure-docker us-docker.pkg.dev
+
+```
+
+
+Tag the Docker image, Push the Docker image, and create a Cloud Run service from the Docker image.
+
+```sh
+export REGION=us-central1
+docker tag $DOCKER_IMAGE us-docker.pkg.dev/$PROJECT/grpcdemo/my-grpc-service
+docker push us-docker.pkg.dev/$PROJECT/grpcdemo/my-grpc-service
+gcloud run deploy my-grpc-service --image us-docker.pkg.dev/$PROJECT/grpcdemo/my-grpc-service --platform managed --use-http2 --allow-unauthenticated --region $REGION
+
+```
+
+Once the deployment completes, store the service URL in an environment variable:
+
+```sh
+CLOUD_RUN_SERVICE_URL=$(gcloud run services describe my-grpc-service --platform managed --region $REGION --format 'value(status.url)' | sed -E 's/http.+\///')
+export CLOUD_RUN_SERVICE_URL
+```
+
+```sh
+grpcurl -import-path $HOME/cloudshell_open/apigee-samples/grpc/grpc-backend/grpc/examples/protos -proto helloworld.proto -d '{"name":"Guest"}' $CLOUD_RUN_SERVICE_URL:443 helloworld.Greeter/SayHello
+
+```
+
+If you get a successful response you’re ready to update the GCP LB to support gRPC.
+
+
+## Update Load Balancer
```

Comment:
This section is pretty complicated.  I think we should wrap this up in a separate script to automate the LB changes.